### DISCUSSION TITLE
MAUI MVVM 中 CheckBox 与命令绑定

### DISCUSSION THREAD

**Discussion by ch-tj-hx-song** [BYJRK/DotNet-Discussions#18] [其他框架]
问一下大佬们，刚接触 `MAUI`，像实现类似 TODO List 的功能。但是无法将 `CheckBox` 与命令绑定，于是使用了 `CommunityToolKit` 中的 `EventToCommandBehavior`，然后在运行过程中报错（无报错信息）

下面是View的问题代码

```xaml
<CheckBox
    Grid.Column="0">
    <CheckBox.Behaviors>
        <toolkit:EventToCommandBehavior
            EventName="CheckedChanged"
            Command="{Binding Source={RelativeSource AncestorType={x:Type viewmodel:MainPageViewModel}}, Path=RemoveTaskCommand}"
            CommandParameter="{Binding .}" />
    </CheckBox.Behaviors>
</CheckBox>
```

这是ViewModel的问题代码（使用了 `CommunityToolKit.Mvvm`）
```csharp
[RelayCommand]
void RemoveTask(string task)
{
    if (Tasks.Contains(task))
    {
        Tasks.Remove(task);
    }
}
```

**Comment by BYJRK** [comment]
我感觉你的 Command 的 Binding 可能有问题，因为 RelativeSource 是从 VisualTree 入手去寻找 Source 的，所以只能找到 MainPage 之类的界面控件，并不能直接找到 ViewModel 类型的实例，所以你可能根本就没有绑到。

**Reply by ch-tj-hx-song** [reply]
那请教一下大佬，在 `DataTemplate` 里怎么绑定到 `ViewModel` 和 `Model` 里的属性呢？

**Reply by BYJRK** [reply]
你的 `DataTemplate` 所对应的控件如果给定了 `DataContext`（如果没有，那就会沿着 Logical Tree 想上找，直到找到一个不为空的 `DataContext`），那么就直接可以绑定了
